ui/src/tree/AzExtTreeDataProvider.ts

```diff
@@ -241,6 +226,21 @@ export class AzExtTreeDataProvider implements IAzExtTreeDataProviderInternal, ty
             }
         }
     }
+
+    private isDuplicateChild(child: AzExtTreeItem, existingChildren: AzExtTreeItem[]): boolean {
+        for (const existingChild of existingChildren) {
```

Comment:
I feel like this is pretty inefficient in terms of having to iterate through an array of children for each single node.  

What if we used an object mapping?  The key value could just be the id, the value could be the node (not really sure if we even need a value if you just need to look it up once to create a unique id)

Reply:
Oh right, I forgot that the existingChildren array is just the duplicates (not the entire array of children).  So this isn't as bad as I thought-- though I think using an object can still make the logic look a little cleaner.

Reply:
Looping through each child here should be the same time complexity as before which used `result.some` but I agree it could be improved. I implemented a map and also accounted for the case where there are multiple duplicate tree items with the same context value. For example:

```ts
treeItem1 = { fullId: "id", contextValue: "a" }
treeItem2 = { fullId: "id", contextValue: "b" }
treeItem3 = { fullId: "id", contextValue: "b" }
```

Before, two of the same uniqueFullIds would have been added to the result array - one for `treeItem2` and another for `treeItem3`. Now it marks `treeItem3` as a duplicate.